### ISSUE TITLE
CSS and JS issue with Filter menu in the bottom Calendar section

### ISSUE THREAD

**Issue by dotnetCarpenter** [ronan-mch/det_sker#311]
Browser: Safari 9.0.3
OS: OS X El Capitan - 10.11.3

When scrolling below the **Calendar** heading, the filter menu jumps to the left, covering the events.

![skaermbillede 2016-02-03 kl 02 54 30](https://cloud.githubusercontent.com/assets/43763/12770643/6d8aee5e-ca22-11e5-946a-7d07a5acd659.png)
*While a top of the **Calendar** heading, everything looks fine*
¨
![skaermbillede 2016-02-03 kl 02 54 03](https://cloud.githubusercontent.com/assets/43763/12770646/6e5f0cfc-ca22-11e5-85ba-dfa4e1ccc90e.png)
*Below the Calander*heading, the menu "jumps" and cover the events.*

While in the danish edition, the filter appear to work, in the English edition it does not. When I click a filter, with numbers, e.g. Food, the event list only display "NO MORE EVENTS".

**Comment by ronan-mch** [comment]
hi Jon,

I can't reproduce this on my Firefox (Linux Mint). There is something weird
happening when I choose a filter with very few events and try to scroll
down. Looking at the html/css can you see anything obvious that might be
causing difficulties?

Cheers,

Ronan

On Wed, Feb 3, 2016 at 3:04 AM, Jon Ege Ronnenberg <notifications@github.com

**Comment by dotnetCarpenter** [comment]
I haven't really looked but it's usually a symptom of changing the
`display` and/or `position` property in CSS. You'll have to look at the
parent element's `position` property. Also make use all HTML elements have
the same [box-sizing](https://css-tricks.com/box-sizing/) property-value.

Bonus info: you'll want to set `[position: sticky)[
http://caniuse.com/#feat=css-sticky]` but support is lacking, so used
`position: fixed` on desktop and do something completely different on
mobile, for now until browser support gets better. Ah, ok might as well
check now....

**Comment by dotnetCarpenter** [comment]
This is the element in question:
```html
<div class="category-filter-affix affix-top" id="sticky-filters">
    <h4>Vis kun</h4>
    <dl id="filter-list" class="col-xs-6 col-sm-12">
  <!-- content removed for brevety --->
</div>
```

`affix-top` top doesn't exist so that's just clutter. When we scroll down, the class change to `affix` which is:
```css
.affix {
  position: fixed;
  top: 0;
}
```

A quick and dirty fix could be:
```css
.affix {
  position: fixed;
  top: 0;
  right: 0;
}
```

But this will just align the filter menu to the right side of the browser window. Probably not what we want.

*Next post will show how to do this proper - but I'll give you a moment to think it over* :jack_o_lantern:

**Comment by dotnetCarpenter** [comment]
Changes need to be done here:
https://github.com/ronan-mch/det_sker/blob/master/app/assets/stylesheets/events.css.scss#L101

I have to go, I'm late for building a house project.... Sounds cool right,
but it is actually just building a house, no collective im afraid.

**Comment by ronan-mch** [comment]
Sorry Jon, I'm a little confused: you suggest `right: 0;` but then you say
>  Probably not what we want.

Do you have a different suggestion? I'm a bit thick when it comes to CSS, so apologies if I'm missing something obvious.

**Comment by Siggik** [comment]
Jeg får samme problem i Safari på en IPad jeg lige lånte.

**Comment by dotnetCarpenter** [comment]
@ronan-mch sorry, haven't had time to do it proper. There is two options, one is to change the html structure and css, another is to add some js magic.
I'm inclined to write some quick and dirty js. But I don't have time right now. Perhaps tomorrow.

If you need it to kinda work now, then just add the `right: 0;`.

Another issue is the js filter is **not** working in english mode. But this should probably be a separate issue.

**Comment by ronan-mch** [comment]
Yeah you're right, the category filter doesn't work in English. No idea
what's going on there, will look at it asap.

lør. 6. feb. 2016 18.57 skrev Jon Ege Ronnenberg <notifications@github.com>:

**Comment by ronan-mch** [comment]
oh jesus it's obvious, I'm not translating the fecking text so of course the filter doesn't work!

**Comment by ronan-mch** [comment]
ok den er nu fikset på dukop.dk, men jeg har vist gjort noget forkert med css right værdien da den hopper når affix bliver aktiveret. ugh :unamused:

**Comment by dotnetCarpenter** [comment]
Nope, du har ikke gjort noget forkert. Du kan prøve at sætte `right: auto;`, men ved ikke om det vil virke i Safari. Hvis det ikke virker så tror jeg at du må finde det script som sætter `affix` klassen og så bruge 

```js
var categoryMenu = $('.category-filter-affix')
categoryMenu.css('right', categoryMenu.position().right)
```
